client/state/state_database.go

```diff
@@ -13,51 +16,75 @@ import (
 /*
 The client has a boltDB backed state store. The schema as of 0.9 looks as follows:
 
-allocations/ (bucket)
-|--> <alloc-id>/ (bucket)
-    |--> alloc -> *structs.Allocation
-    |--> alloc_runner persisted objects (k/v)
-	|--> <task-name>/ (bucket)
-        |--> task_runner persisted objects (k/v)
-
-devicemanager/
-|--> plugin-state -> *dmstate.PluginState
+meta/
+|--> version -> "1"
+|--> upgraded -> time.Now().Format(timeRFC3339)
+allocations/
+|--> <alloc-id>/
+   |--> alloc         -> allocEntry{*structs.Allocation}
+   |--> deploy_status -> deployStatusEntry{*structs.AllocDeploymentStatus}
+   |--> task-<name>/
+      |--> local_state -> *trstate.LocalState # Local-only state
+      |--> task_state  -> *structs.TaskState  # Sync'd to servers
+
+devices/
+|--> plugin_state -> *dmstate.PluginState
 */
 
 var (
+	// metaBucketName is the name of the metadata bucket
+	metaBucketName = []byte("meta")
+
+	// metaVersionKey is the key the state schema version is stored under.
+	metaVersionKey = []byte("version")
+
+	// metaVersion is the value of the state schema version to detect when
+	// an upgrade is needed. It skips the usual boltdd/msgpack backend to
+	// be as portable and futureproof as possible.
+	metaVersion = "1"
```

Comment:
❓(nit-ish) Should we one index versions and use a version of `"2"` for this state? - and then have the current state be nominally "1".